controllers/front/CartController.php

```diff
@@ -326,12 +353,23 @@ protected function processChangeProductInCart()
         }
 
         // Check product quantity availability
+        $ErrorKey = $isUpdateOperationDown ? 'updateOperationError' : 'errors';
         if ($this->id_product_attribute) {
             if (!Product::isAvailableWhenOutOfStock($product->out_of_stock) && !Attribute::checkAttributeQty($this->id_product_attribute, $qty_to_check)) {
-                $this->errors[] = $this->trans('There are not enough products in stock', array(), 'Shop.Notifications.Error');
+                array_push($this->{$ErrorKey},
+                    $this->trans(
+                        'There are not enough products in stock',
+                        array(),
+                        'Shop.Notifications.Error')
+                );
             }
         } elseif (!$product->checkQty($qty_to_check)) {
-            $this->errors[] = $this->trans('There are not enough products in stock', array(), 'Shop.Notifications.Error');
+            array_push($this->{$ErrorKey},
+                $this->trans(
+                    'There are not enough products in stock',
+                    array(),
+                    'Shop.Notifications.Error')
+            );
```

Comment:
Simplified statement would be :
```php
if ((
        $this->id_product_attribute
        && !Product::isAvailableWhenOutOfStock($product->out_of_stock)
        && !Attribute::checkAttributeQty($this->id_product_attribute, $qty_to_check)
    )
    ||
    !$product->checkQty($qty_to_check)
) {
    array_push(
        $this->{$ErrorKey},
        $this->trans(
            'There are not enough products in stock',
            array(),
            'Shop.Notifications.Error'
        )
    );
}
```
But this is still a bit complicated. Maybe a private method wrapping all the "ifs" would be cleaner. Something like :
```php
if ($this->availabilityErrorShouldBeRaised($product, $qty_to_check) {
    array_push(
        /*...*/
    );
}
```

Reply:
@LittleBigDev let's make it simple, take an exemple:
`if (x) { if (y) {return 'toto'; } } esleif (z) { return 'coco'; }`
let's test the following values: x=1, y=0, z=1
we will get **nothing** as result
but if we adopt your suggestion:
`if ((x&y) || z) { return 'toto' }`
then we will get **'toto'** as result 
do you understand the main problem? we need to consider all cases :wink: 
so we can not simplified it.

Reply:
Yes my bad, I didn't double check my suggestion.
Anyway, you masked it in a method and I'm fine with it.